### DISCUSSION TITLE
关于USB与网络控制目前的发现

### DISCUSSION THREAD

**Discussion by SheepYhangCN** [ZiHaoSaMa66/OsEasy-ToolBox#31] [General]
（测试版本v10.9.1）
分别由```easyusbflt.sys```和```OeNetLimit.sys```控制
且一旦开启 即使删除驱动也不会关掉
需要在教师端开启前就要删除驱动
也就是说 一旦被控 就要重启电脑
在连上教师端前拔网线 删除驱动再插回去
（或拼手速 在连上教师机前退出噢易或禁用网络适配器）

目前起效果的尝试是文件粉碎删掉这两个文件
和```x64/DeviceControl_x64.exe```
再```sc delete```掉这两个服务
所以我不确定是删除服务 删除文件 还是删除DeviceControl起效果了

**Comment by FarryX** [comment]
您好，在@ZiHaoSaMa66 的oe工具箱中，我的学校oe版本，（未开启usb管控）在教师开启网络管控后，使用工具箱的理论解除网络管控功能，可以有效，但如果教师提前，同时打开了usb管控和网络管控，形成闭环，我无法打开u中工具箱，也无法连网下载，那我该如何手动解除网络管控呢，粉碎mutclient.sys就可以了吗，谢谢

**Comment by ZiHaoSaMa66** [comment]
我目前使用的方法是
关闭根服务保护后
重命名 x64/DeviceControl_x64.exe
 接着重启机器 就可以解除管控了
(如果设定为每次重启都还原机器则不行)

**Reply by FarryX** [reply]
学校设置了每次重启会还原配置，所以每次遇到网络管控时，只有当侥幸没有开启usb管控，从u盘打开工具箱，关闭网络管控才行。如果usb管控和网络管控同时开启，理论上我无法有任何方法再解除网络管控吗。请问工具箱解除原理是什么，直接循环删除网络管控的dll文件吗，因为工具箱在解除网络管控时没有重启，所以很好奇，手动能不能复刻出工具箱的解除网络管控，如果可以，至少在两种管控都开启时，可以重新联网下载工具箱。

**Comment by ZiHaoSaMa66** [comment]
如果只是单独解除网络是可以的
具体可以查看源码中的cmd脚本实现
在src/remain.py文件中（有点乱w
建议看dev分支的代码 稍微整理了一下代码
dll解除可以问问ai应该怎么调用

**Comment by ZiHaoSaMa66** [comment]
直接sc stop停止掉噢易的网络限制服务应该就可以了

**Reply by FarryX** [reply]
好的感谢，今天试试这个方法

**Reply by ZiHaoSaMa66** [reply]
> > 直接sc stop停止掉噢易的网络限制服务应该就可以了
> 
> 是不是OeNetLimit服务？我用cmd sc stop OeNetLimit 和 MMPC，taskill Student.exe，DevicesControl_64.exe，关闭它们后并注销，网络限制不会被解除

那可能又进化了（？） 旧的实现当中执行完stop 服务这一步不需要注销就已经解除限制了

注意一下顺序再试试呢
先stop mmpc 再k掉学生端本体和DevicesControl_64 最后再停止网络限制服务

**Reply by NYSkyfox** [reply]
> > > 直接sc stop停止掉噢易的网络限制服务应该就可以了
> > 
> > 
> > 是不是OeNetLimit服务？我用cmd sc stop OeNetLimit 和 MMPC，taskill Student.exe，DevicesControl_64.exe，关闭它们后并注销，网络限制不会被解除
> 
> 那可能又进化了（？） 旧的实现当中执行完stop 服务这一步不需要注销就已经解除限制了
> 
> 注意一下顺序再试试呢 先stop mmpc 再k掉学生端本体和DevicesControl_64 最后再停止网络限制服务

好的谢谢，成功了
